types/changelog-parser/index.d.ts

```diff
@@ -1,25 +1,68 @@
-// Type definitions for changelog-parser 2.7
+// Type definitions for changelog-parser 2.8
 // Project: https://github.com/hypermodules/changelog-parser
-// Definitions by:  Adam Zerella <https://github.com/adamzerella>
+// Definitions by: Adam Zerella <https://github.com/adamzerella>
+//                 Oliver Steele <https://github.com/osteele>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
-// TypeScript Version: 2.2
+// TypeScript Version: 4.4
 
+// Options must contain exactly one of filePath or text. This can be expressed
+// with a type, but not an interface. (See
+// <https://github.com/osteele/vscode-p5server/blob/fee241c06f/scripts/types/changelog-parser.d.ts>,
+// which contains the ambient interface that this type was extracted from.)
+//
+// The DefinitelyTyped lint settings don't allow type literals, so define an
+// interface here, and use it to construct a type expression in the argument.
+
+// This type can't be exported, because the module interface has a default
+// export.
+//
+// To use the type of this interface in your code, import `parseChangelog` and
+// define:
+//   type Options = Exclude<Parameters<typeof parseChangelog>[0], string>;
 interface Options {
     /**
      * Path to changelog file.
      */
     filePath: string;
+    /**
+     * Text of changelog file (you can use this instead of filePath).
+     */
+    text: string;
     /**
      * Removes the markdown markup from the changelog entries by default.
      * You can change its value to false to keep the markdown.
      */
     removeMarkdown: boolean;
 }
 
+// This type can't be exported, because the module interface has a default
```

Comment:
so, if you want to export the `Changelog` interface, you can use (declaration merging)[https://www.typescriptlang.org/docs/handbook/declaration-merging.html#merging-namespaces] to have a namespace declaring the types, with the same name of the function, like so:
```ts
declare namespace parseChangelog {
    interface Options { ... }
    interface Changelog { ... }
}
declare function parseChangelog();
export = parseChangelog;
```
then you can refer to the types when importing from another file, like so:
```ts
import changelog from "parseChangelog";
changelog(...);
var opt: changelog.Options;
var log: changelog.Changelog;
```

Reply:
Thank you, I have revised the PR to use your recommendation.